### PR TITLE
Update marketing website link in monorepo

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Not my first batch change!

This PR updates `about.sourcegraph.com` to `sourcegraph.com` as part of the marketing website migration.

[_Created by Sourcegraph batch change `bolaji.olajide/marketing-website-link-update`._](https://sourcegraph.sourcegraph.com/users/bolaji.olajide/batch-changes/marketing-website-link-update)

### Test plan

Link update.
Manual review

### CODE REVIEW

**Review by unknwon — APPROVED** [review]
As long as the CI is happy 😂

**Comment by iskyOS** [comment]
Sorry for the long delay! This has been on my to-do to spend time looking through. As a quick solution, I did update a few links in sourcegraph.com (the most critical ones) to use sourcegraph.com last week.

@unknwon Is there anyone on the code search team who should also review this to make sure it doesn't break anything? I'm nervous about such a large change to production Sourcegraph.

**Comment by unknwon** [comment]
@iskyOS - I expect this PR is only doing find-and-replace for string literals, which is pretty safe in this case because about.sourcegraph.com is a very specific string (as long as CI is happy).

The only thing I'd be worried about is... whether all of those links now pointing to sourcegraph.com, have actually been serving content like they do on about.sourcegraph.com.

File: cmd/frontend/internal/app/app.go
Comment (iskyOS) [inline cmd/frontend/internal/app/app.go:1]: @unknwon This /go route is missing but it only redirects to the blog. I honestly am not sure what it is meant to do in this context

File: dev/check/broken-urls.bash
Comment (iskyOS) [inline dev/check/broken-urls.bash:1]: I am not sure what this code is doing or how this change will impact it.

File: doc/code_search/how-to/opengrok.md
Comment (iskyOS) [inline doc/code_search/how-to/opengrok.md:1]: Both of these links (before and after) appear to be dead

File: internal/database/event_logs_test.go
Comment (iskyOS) [inline internal/database/event_logs_test.go:1]: Route missing, not sure what this does?

File: internal/repos/testdata/sources/GITHUB-LIST-REPOS/graphql-fallback.yaml
Comment (iskyOS) [inline internal/repos/testdata/sources/GITHUB-LIST-REPOS/graphql-fallback.yaml:1]: I am not sure what changed here but it makes me a little nervous looking at

File: doc/_resources/assets/redirects
Comment (iskyOS) [inline doc/_resources/assets/redirects:1]: /retrospectives and /retrospectives/ routes missing

**Comment by iskyOS** [comment]
> @iskyOS - I expect this PR is only doing find-and-replace for string literals, which is pretty safe in this case because about.sourcegraph.com is a very specific string (as long as CI is happy).
> 
> The only thing I'd be worried about is... whether all of those links now pointing to sourcegraph.com, have actually been serving content like they do on about.sourcegraph.com.

Good point. I just went through and testing everything. There are a few missing routes, and some changes that look odd. I'll open a PR for new routes.

File: cmd/frontend/internal/app/app.go
Comment (unknwon) [reply cmd/frontend/internal/app/app.go:1]: This is an very old URL, we used to live blogging GopherCon, which we have stopped doing for some years. We can delete for now and won't lose any.

File: dev/check/broken-urls.bash
Comment (unknwon) [reply dev/check/broken-urls.bash:1]: I think it should work as before, it is finding all links to Sourcegraph.com and making real requests to make sure all those links is correctly served.

File: doc/code_search/how-to/opengrok.md
Comment (unknwon) [reply doc/code_search/how-to/opengrok.md:1]: Yeah, I think we can delete it :joy:

File: doc/dev/how-to/documentation_implementation.md
[the remark this answers is not in the document]
Comment (unknwon) [reply doc/dev/how-to/documentation_implementation.md:1]: Looks like the correct link is https://handbook.sourcegraph.com/departments/engineering/dev/process/deployments/playbooks/#restarting-docssourcegraphcom? which means the old link was also not working due to changes happened on the handbook side.

File: internal/codemonitors/background/testdata/TestEmail/one_result_with_results/html.golden
[the remark this answers is not in the document]
Comment (unknwon) [reply internal/codemonitors/background/testdata/TestEmail/one_result_with_results/html.golden:1]: It's testdata, I think it's fine if it's missing in the real site.

File: internal/database/event_logs_test.go
Comment (unknwon) [reply internal/database/event_logs_test.go:1]: It's testdata, I think it's fine if it's missing in the real site.

File: internal/repos/testdata/sources/GITHUB-LIST-REPOS/graphql-fallback.yaml
Comment (unknwon) [reply internal/repos/testdata/sources/GITHUB-LIST-REPOS/graphql-fallback.yaml:1]: It's testdata, I think it's fine if it's missing in the real site.

**Comment by iskyOS** [comment]
@unknwon I've gone through all of the changes + comments and this looks good to me now! Can you help me resolve the merge conflicts? I've never done that outside the UI

**Pull request merged**